### PR TITLE
Content Uptime

### PR DESCRIPTION
**Pull request by ezequielc**
Common Template will listen for Uptime requests from Player and in turn request the status for every Rise component to report their statuses.

A Template will report uptime as `error` whenever one or more components are not responding or in error state. Also, a utility method `RisePlayerConfiguration.ContentUptime.setUptimeError() ` was added so that Template designers can programmatically specify an uptime error state.

@alex-deaconu Please review. Thanks!

### CODE REVIEW

File: test/unit/rise-content-uptime.test.js [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+    sinon.stub( RisePlayerConfiguration.Helpers, "getRiseElements", function() {
+      return Object.values( expectedComponents );
+    });
+
+  });
+
+  afterEach( function() {
+    clock.restore();
+
+    window.addEventListener.restore();
```
Comment (alex-deaconu) [inline test/unit/rise-content-uptime.test.js]: @ezequielc you can just run `sinon.restore()` to restore all mocks here (including `clock` I believe). Thanks.

**Comment by alex-deaconu** [comment]
@ezequielc looks good, one suggestion inline. Thanks!

File: test/unit/rise-content-uptime.test.js (hunk 1 above)
Comment (ezequielc) [reply test/unit/rise-content-uptime.test.js]: @alex-deaconu I tried that but stubs were not being restored. Maybe it is something particular with the lib version. Thanks

Comment (alex-deaconu) [reply test/unit/rise-content-uptime.test.js]: @ezequielc you're right. This uses the old version of `sinon`. However, if you bump the `widget-tester` version it should work correctly. Thanks!

**Comment by ezequielc** [comment]
@alex-deaconu Please review my latest commit. I changed the order of firing event to components and setting the timeout as I was experiencing receiving all results before setting the timeout, which would cause the timeout to fire even after receiving all components results and report and additional and incorrect uptime response to players. Thanks!

**Comment by alex-deaconu** [comment]
@ezequielc reviewed, looks good 👍

File: test/unit/rise-content-uptime.test.js (hunk 1 above)
Comment (ezequielc) [reply test/unit/rise-content-uptime.test.js]: @alex-deaconu Ok, I will give another shot. Thanks

Comment (ezequielc) [reply test/unit/rise-content-uptime.test.js]: @alex-deaconu I managed to get it to work with sandbox. Please review latest commit. Thanks!

Comment (alex-deaconu) [reply test/unit/rise-content-uptime.test.js]: @ezequielc looks great! Thanks for revising.

**Pull request merged**